### PR TITLE
Initial MAD PVP Support

### PR DESCRIPTION
**Pull request by TurtIeSocks**
- Uses Ohbem to calculate pvp ranks for each request
- Configurable leagues
- Configurable level caps
- Requires config updates!

Before pulling this commit please be sure to set these config settings, specifically the leagues array, even if you aren't planning to have ReactMap calculate your PVP ranks:
```js
  "database": {
    "settings": {
      "reactMapHandlesPvp": false,
      "leagues": [
        { "name": "great", "cp": 1500 },
        { "name": "ultra", "cp": 2500 }
      ],
      "pvpLevels": [50, 51]
    },
  }
```

### CODE REVIEW

File: server/src/models/Pokemon.js [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
 const getAreaSql = require('../services/functions/getAreaSql')
 
+const leagueObj = {}
+leagues.forEach(league => leagueObj[league.name] = league.cp)
+const ohbem = reactMapHandlesPvp ? new Ohbem({
+  leagues: leagueObj,
+  pokemonData: masterfile,
+  levelCaps: pvpLevels,
+  cachingStrategy: Ohbem.cachingStrategies.memoryHeavy,
+}) : undefined
```
Comment (Mygod) [inline server/src/models/Pokemon.js:22]: `null` might be better. 🤔

**Review by Mygod — APPROVED** [review]
Ohbem claims another victim. Hype.

File: server/src/configs/default.json [hunk 2]
```diff
@@ -201,7 +201,9 @@
     "settings": {
       "userTableName": "users",
       "sessionTableName": "session",
-      "leagues": ["great", "ultra"]
+      "leagues": [],
+      "reactMapHandlesPvp": false,
+      "pvpLevels": []
```
Comment (Mygod) [inline server/src/configs/default.json:206]: Why empty for default?

**Review by Mygod — COMMENTED** [review]
Little cup does not work.

**Comment by versx** [comment]
> Little cup does not work.

:eyes: Why you make little cup no work?

**Comment by Mygod** [comment]
> > Little cup does not work.
> 
> 👀 Why you make little cup no work?

Cuz heaven forbids this map has a separate master file.

**Comment by versx** [comment]
> > > Little cup does not work.
> > 
> > 
> > 👀 Why you make little cup no work?
> 
> Cuz heaven forbids this map has a separate master file.

😭 Maybe some day.

File: server/src/configs/default.json (hunk 2 above)
Comment (TurtIeSocks) [reply server/src/configs/default.json:206]: So admins can choose what they want to track.

**Comment by TurtIeSocks** [comment]
Little cup has been fixed.

File: server/src/configs/default.json (hunk 2 above)
Comment (Mygod) [reply server/src/configs/default.json:206]: Default should provide a reasonable default, probably not empty. 🤔

**Pull request merged**